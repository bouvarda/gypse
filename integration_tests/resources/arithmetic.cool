
class Main() {
    def test_integer(): Unit = {
      var a: Int = (10 * (4 / 2)) + 30 - 8;
      out_int(a);
      out_nl();
      if ((a > 0) && (a <= 10)) {
          out_string("Between 1 and 10")
      } else if (a <= 50) {
          out_string("Between 11 and 50")
      } else {
          out_string("Above 50")
      };
      out_nl();
      if ((a < 0) || (a > 100)) {
          out_string("'a' is invalid")
      } else {
          out_string("'a' is valid")
      };
      out_nl();
    };

    def test_negative_numbers(): Unit = {
        var i: Int = -1;
        out_int(i * i * i);
        out_nl();
    };

    def test_float(): Unit = {
        var f: Float = 0.1;
        out_float(f * 0.2, 2);
        out_nl();
        out_float(f * -f, 4);
        out_nl();
        out_float(3.1415 / 10.0, 3);
        out_nl();
        out_float(f + 123.4, 1);
        out_nl();
        if ((f - 0.5) > 0.0) {
            out_string("wrong")
        } else {
            out_string("f is negative")
        };
    };

    {
        test_integer();
        test_negative_numbers();
        test_float();
    };
}