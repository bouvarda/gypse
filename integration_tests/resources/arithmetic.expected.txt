42
Between 11 and 50
'a' is valid
-1
0.02
-0.0100
0.314
123.5
f is negative